libraries/ms-common/inc/spi.h

```diff
@@ -0,0 +1,76 @@
+// HAL SPI
+
+// Interface for SPI; used to send and receive messages according to the SPI specification.
+
+// To use, fill a SPISettings struct with the desired settings and call spi_init once.
+// You cannot change these settings later.
+// After initializing a SPI peripheral, you can use the spi_exchange and spi_set_cs_state
+// functions. You must initialize each peripheral individually.
+
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "gpio.h"
+#include "status.h"
+
+typedef enum {
+  SPI_PERIPH_1 = 0,
+  SPI_PERIPH_2,
```

Comment:
I would add another entry such as `SPI_NUM_PERIPHS` or something similar so we can keep track of how many interfaces are supported without needing to update it manually since enums auto-increment.

Along the same lines, I would personally move this enum to a platform-specific header so we aren't locked to 2 SPI ports. For example, we aren't currently planning on supporting SPI on x86.